### PR TITLE
Universal implementation SPA

### PR DESCRIPTION
**Pull request by JoelRoxell**
### Description

Allows the application to be rendered server-side with CSS-modules and use webpack aliases to resolve files.
### Related issues
#6

### CODE REVIEW

File: src/server/style-import-hook.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { readFileSync } from 'fs';
+
+let result;
+
+// Require hook
+function hook(compile, extension) {
+  require.extensions[extension] = function(m, filename) {
+    const tokens = compile(filename);
+
+    let moduleStyles = m._compile('module.exports = ' + JSON.stringify(tokens), filename);
```
Comment (LinusLjung) [inline src/server/style-import-hook.js]: Use `const` or return right away

File: src/server/style-import-hook.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    return moduleStyles;
+  };
+}
+
+// Load CSS modules and get nodes tokens
+function fetch(filename) {
+  let css = stylus.render(readFileSync(filename, 'utf8'), { filename });
```
Comment (LinusLjung) [inline src/server/style-import-hook.js]: `const`

**Review by LinusLjung — CHANGES_REQUESTED** [review]

**Review by widenanders — APPROVED** [review]

File: src/config/index.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
+  TITLE: 'snabb',
+  DESCRIPTION: 'Barebone application renderd server-side, react, react-router, CSS-modules.',
+
+  MOUNTING_POINT: 'app',
+
+  files: {
```
Comment (zappen999) [inline src/config/index.js:7]: Should be in caps as well

File: src/config/index.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    STYLE_BUNDLE: 'main.css',
+    CLIENT_BUNDLE: 'bundle.js'
+  },
+
+  entry: {
```
Comment (zappen999) [inline src/config/index.js:12]: Caps

File: src/server/style-import-hook.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import Scope from 'postcss-modules-scope';
+import Parser from 'postcss-modules-parser';
+import genericNames from 'generic-names';
+import stylus from 'stylus';
+import { readFileSync } from 'fs';
+
+let result;
+
+// Require hook
+function hook(compile, extension) {
```
Comment (zappen999) [inline src/server/style-import-hook.js]: Proper docs would be nice. Especially when there is so much black magic going on in this project 😄

File: webpack.config.js [hunk 6]
```diff
@@ -24,15 +35,30 @@ config.module.loaders = [
...
+    test: /\.styl$/,
+    loader: ExtractTextPlugin.extract('css-loader?modules&camelCase&localIdentName=[name]__[local]___[hash:base64:5]!stylus-loader') // eslint-disable-line
   }
 ];
 
 config.plugins = [
   new webpack.DefinePlugin({
     'process.env': {
-      'NODE_ENV': JSON.stringify('production')
+      NODE_ENV: JSON.stringify('development')
```
Comment (zappen999) [inline webpack.config.js:48]: stringify necessary?

File: src/server/style-import-hook.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import postcss from 'postcss';
+import Values from 'postcss-modules-values';
+import LocalByDefault from 'postcss-modules-local-by-default';
+import ExtractImports from 'postcss-modules-extract-imports';
+import Scope from 'postcss-modules-scope';
+import Parser from 'postcss-modules-parser';
+import genericNames from 'generic-names';
+import stylus from 'stylus';
+import { readFileSync } from 'fs';
```
Comment (zappen999) [inline src/server/style-import-hook.js:9]: Inconsistent casing

File: webpack.config.js (hunk 6 above)
Comment (JoelRoxell) [reply webpack.config.js:48]: Probably not, but I'll check this when the focus is on project compilation and automation(another issue).

File: src/server/style-import-hook.js (hunk 7 above)
Comment (JoelRoxell) [reply src/server/style-import-hook.js:9]: Don't blame me man, https://nodejs.org/api/fs.html#fs_fs_readfilesync_file_options.

**Review by LinusLjung — APPROVED** [review]

File: src/server/style-import-hook.js (hunk 5 above)
Comment (JoelRoxell) [reply src/server/style-import-hook.js]: yes

**Pull request merged**